### PR TITLE
Add MinimalTlsVersion Property to MI and Sql DB APIs

### PR DESCRIPTION
**Pull request by ziwa-msft**
Test generation is pending our Prod deployment. ETA 2/28

### CODE REVIEW

**Comment by ziwa-msft** [comment]
@erich-wang Also want to commen that running generate.ps1 is broken now as it keeps trying to pass in the latest for verison parameter when calling autorest. I have to run the following cmd manually:

cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/readme.md --csharp --reflect-api-versions --csharp-sdks-folder=.

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SdkInfo_SqlManagementClient.cs [hunk 1]
```diff
@@ -120,9 +120,11 @@ internal static partial class SdkInfo
       public static readonly String GithubForkName = "Azure";
       public static readonly String GithubBranchName = "master";
       public static readonly String GithubCommidId = "dcd51ecb0bba6cd98f42383f0ef0159efc64a658";
-      public static readonly String CodeGenerationErrors = "";
+      public static readonly String CodeGenerationErrors = ""
```
Comment (isra-fel) [inline sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SdkInfo_SqlManagementClient.cs]: This is blocking CI

**Comment by isra-fel** [comment]
> @erich-wang Also want to commen that running generate.ps1 is broken now as it keeps trying to pass in the latest for verison parameter when calling autorest. I have to run the following cmd manually:

Thanks for the feedback! Could you share more info about how generate.ps1 was broken? What was the error?

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SdkInfo_SqlManagementClient.cs (hunk 1 above)
Comment (ziwa-msft) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Generated/SdkInfo_SqlManagementClient.cs]: Thanks! will fix this.

**Comment by ziwa-msft** [comment]
> > @erich-wang Also want to commen that running generate.ps1 is broken now as it keeps trying to pass in the latest for verison parameter when calling autorest. I have to run the following cmd manually:
> 
> Thanks for the feedback! Could you share more info about how generate.ps1 was broken? What was the error?

error is: 

Write-ErrorLog : FATAL: System.InvalidOperationException: Found incompatible property types ,  for property 'type' in schema inheritance chain
   at AutoRest.Modeler.SchemaResolver.ExpandAllOf(Schema schema) in /opt/vsts/work/1/s/src/SchemaResolver.cs:line 157
   at AutoRest.Modeler.SchemaResolver.Unwrap(Schema schema) in /opt/vsts/work/1/s/src/SchemaResolver.cs:line 75
   at AutoRest.Modeler.SchemaBuilder.BuildServiceType(String serviceTypeName, Boolean required) in /opt/vsts/work/1/s/src/SchemaBuilder.cs:line 35
   at AutoRest.Modeler.SwaggerModeler.BuildCompositeTypes() in /opt/vsts/work/1/s/src/SwaggerModeler.cs:line 348
   at AutoRest.Modeler.SwaggerModeler.Build(ServiceDefinition serviceDefinition) in /opt/vsts/work/1/s/src/SwaggerModeler.cs:line 66
   at AutoRest.Modeler.Program.<ProcessInternal>d__2.MoveNext() in /opt/vsts/work/1/s/src/Program.cs:line 60
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at NewPlugin.<Process>d__15.MoveNext()
  Error: Plugin imodeler1 reported failure.
At C:\Users\ziwa\Documents\WindowsPowerShell\Modules\CodeGenerationModules\AutoRestCodeGenerationModule\AutoRestCodeGenerationModule.psm1:104 char:5
+     Write-ErrorLog $stderr
+     ~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
    + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Write-ErrorLog

**Comment by ziwa-msft** [comment]
> > @erich-wang Also want to commen that running generate.ps1 is broken now as it keeps trying to pass in the latest for verison parameter when calling autorest. I have to run the following cmd manually:
> 
> Thanks for the feedback! Could you share more info about how generate.ps1 was broken? What was the error?

It is due the fact that latest is passed in as an parameter when running the script for --version:

cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/readme.md --csharp --version=latest --reflect-api-versions --csharp-sdks-folder=.

I unblocked myself by running without --version=latest

**Comment by cicovica** [comment]
It looks like the build is still failing:

Tried to execute:
```
msbuild build.proj /t:Build /p:Scope=SqlManagement
```

Got the following error:
```
"D:\AzureGit\ziwa\azure-sdk-for-net\build.proj" (Build target) (1) ->
"D:\AzureGit\ziwa\azure-sdk-for-net\eng\mgmt.proj" (default target) (2) ->
"D:\AzureGit\ziwa\azure-sdk-for-net\sdk\SqlManagement\Microsoft.Azure.Management.SqlManagement\src\Microsoft.Azure.Management.Sql.csproj" (default target) (3:27) ->
"D:\AzureGit\ziwa\azure-sdk-for-net\sdk\SqlManagement\Microsoft.Azure.Management.SqlManagement\src\Microsoft.Azure.Management.Sql.csproj" (Build target) (3:29) ->
  Generated\SqlManagementClient.cs(493,24): error CS0246: The type or namespace name 'ILongTermRetentionManagedInstanceBackupsOperations' could not be found (are you missing a using directive or an assembly reference?) [D:\AzureGit\ziwa\azure-sdk-for-net\sdk\SqlManagement\Microsof
t.Azure.Management.SqlManagement\src\Microsoft.Azure.Management.Sql.csproj]
  Generated\SqlManagementClient.cs(498,24): error CS0246: The type or namespace name 'IManagedInstanceLongTermRetentionPoliciesOperations' could not be found (are you missing a using directive or an assembly reference?) [D:\AzureGit\ziwa\azure-sdk-for-net\sdk\SqlManagement\Microso
ft.Azure.Management.SqlManagement\src\Microsoft.Azure.Management.Sql.csproj]
  Generated\SqlManagementClient.cs(503,24): error CS0246: The type or namespace name 'IWorkloadGroupsOperations' could not be found (are you missing a using directive or an assembly reference?) [D:\AzureGit\ziwa\azure-sdk-for-net\sdk\SqlManagement\Microsoft.Azure.Management.SqlMan
agement\src\Microsoft.Azure.Management.Sql.csproj]
  Generated\SqlManagementClient.cs(508,24): error CS0246: The type or namespace name 'IWorkloadClassifiersOperations' could not be found (are you missing a using directive or an assembly reference?) [D:\AzureGit\ziwa\azure-sdk-for-net\sdk\SqlManagement\Microsoft.Azure.Management.S
qlManagement\src\Microsoft.Azure.Management.Sql.csproj]
  Generated\SqlManagementClient.cs(30,84): error CS0535: 'SqlManagementClient' does not implement interface member 'ISqlManagementClient.ManagedDatabaseRestoreDetails' [D:\AzureGit\ziwa\azure-sdk-for-net\sdk\SqlManagement\Microsoft.Azure.Management.SqlManagement\src\Microsoft.Azur
e.Management.Sql.csproj]
```

**Comment by cicovica** [comment]
@isra-fel Can you advise on why the generated code has errors, preventing the build to be successful?

**Comment by markcowl** [comment]
You likely want to usee --version-V2 when running this - this is what the SDK automation is currnetly doing, it uses the 2.x version fo autorest-core.

**Comment by ziwa-msft** [comment]
> You likely want to usee --version-V2 when running this - this is what the SDK automation is currnetly doing, it uses the 2.x version fo autorest-core.

Hi Mark, 

Are you suggesting us to run this:

PS E:\Github2\azure-sdk-for-net\sdk\sqlmanagement\Microsoft.Azure.Management.SqlManagement\src> cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/readme.md --csharp --reflect-api-versions --csharp-sdks-folder=.  --version-V2

I have kicked this off. Hopefully it solves the issue. But for some reason each iteration is taking ~3 hours which is not expected.

**Comment by ziwa-msft** [comment]
> > You likely want to usee --version-V2 when running this - this is what the SDK automation is currnetly doing, it uses the 2.x version fo autorest-core.
> 
> Hi Mark,
> 
> Are you suggesting us to run this:
> 
> PS E:\Github2\azure-sdk-for-net\sdk\sqlmanagement\Microsoft.Azure.Management.SqlManagement\src> cmd.exe /c autorest.cmd https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/readme.md --csharp --reflect-api-versions --csharp-sdks-folder=. --version-V2
> 
> I have kicked this off. Hopefully it solves the issue. But for some reason each iteration is taking ~3 hours which is not expected.

Just synced with another team member and he suggested to use the following:

Start-AutoRestCodeGeneration -ResourceProvider "sql/resource-manager" -SdkRepoRootPath "." -AutoRestVersion "2.0.4413"

Retrying now.

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/eng/mgmt/mgmtmetadata/sql_resource-manager.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿Installing AutoRest version: 2.0.4413
```
Comment (isra-fel) [inline sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/eng/mgmt/mgmtmetadata/sql_resource-manager.txt]: This file should be in /eng/mgmt/mgmtmetadata

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Microsoft.Azure.Management.Sql.csproj [hunk 3]
```diff
@@ -7,16 +7,13 @@
 		<PackageId>Microsoft.Azure.Management.Sql</PackageId>
 		<Description>Azure SQL Management SDK library</Description>
 		<AssemblyName>Microsoft.Azure.Management.Sql</AssemblyName>
-		<Version>1.39.0-preview</Version>
+		<Version>1.40.0-preview</Version>
```
Comment (isra-fel) [inline sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Microsoft.Azure.Management.Sql.csproj]: Except for csproj, please also update version in AssemblyInfo.cs

**Review by isra-fel — CHANGES_REQUESTED** [review]
Hi @ziwa-msft , two minor issues left. Thanks

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/eng/mgmt/mgmtmetadata/sql_resource-manager.txt (hunk 2 above)
Comment (cicovica) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/eng/mgmt/mgmtmetadata/sql_resource-manager.txt]: Updated, please review

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Microsoft.Azure.Management.Sql.csproj (hunk 3 above)
Comment (cicovica) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/Microsoft.Azure.Management.Sql.csproj]: Updated, please review

File: sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/eng/mgmt/mgmtmetadata/sql_resource-manager.txt (hunk 2 above)
Comment (isra-fel) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/eng/mgmt/mgmtmetadata/sql_resource-manager.txt]: I think it's accidentally removed...

Comment (cicovica) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/eng/mgmt/mgmtmetadata/sql_resource-manager.txt]: That is weird, I moved it to /eng/mgmt/mgmtmetadata but it's not showing up in the diff here

Comment (cicovica) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/eng/mgmt/mgmtmetadata/sql_resource-manager.txt]: Updated

Comment (cicovica) [reply sdk/sqlmanagement/Microsoft.Azure.Management.SqlManagement/src/eng/mgmt/mgmtmetadata/sql_resource-manager.txt]: Please review again

**Review by isra-fel — APPROVED** [review]
LGTM thanks

**Pull request merged**